ory_mocking/Testservers/testservers.go

```diff
@@ -0,0 +1,131 @@
+package Testservers
+
+import (
+	"fmt"
+	handlers "identity_platform_login_ui/ory_mocking/Handlers"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"os"
+)
+
+var schema_server_url string
+var kratos *httptest.Server
+var hydra *httptest.Server
+var schema *httptest.Server
+
+func createKratosMockServer() *httptest.Server {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/self-service/errors", handlers.SelfServiceErrorsHandler)
+	mux.HandleFunc("/sessions/whoami", handlers.SessionWhoAmIHandler)
+	mux.HandleFunc("/self-service/login/browser", handlers.SelfServiceLoginBrowserHandler)
+	mux.HandleFunc("/self-service/login", handlers.SelfServiceLoginHandler)
+
+	s := httptest.NewServer(mux)
+	os.Setenv("KRATOS_PUBLIC_URL", s.URL)
+	return s
+}
+func createHydraMockServer() *httptest.Server {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/admin/oauth2/auth/requests/login/accept", handlers.Oauth2AuthRequestLoginAcceptHandler)
+	mux.HandleFunc("/admin/oauth2/auth/requests/consent", handlers.Oauth2AuthRequestConsentHandler)
+	mux.HandleFunc("/admin/oauth2/auth/requests/consent/accept", handlers.Oauth2AuthRequestConsentAcceptHandler)
+	s := httptest.NewServer(mux)
+	os.Setenv("HYDRA_ADMIN_URL", s.URL)
+	return s
+}
+
+func createSchemaMockServer() *httptest.Server {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/testschema", SchemaHandler)
+	s := httptest.NewServer(mux)
+	schema_server_url = s.URL
+	handlers.SetSchemaServerURL(s.URL)
+	handlers.SetSchemaServerURL(s.URL)
+	return s
+}
```

Comment:
Why is this needed?

Reply:
The client sometimes receives an object that includes an object, a schema ID, and a schema URL. The client supposedly (according to documentation) validates the object structure against the schema URL. During debugging when some of the error messages weren't clear I implemented this schema server, and passed it's url into the http responses from the mock server.

Reply:
I checked and the client doesn't actually validate with a schema received from the url. I'll take the schemaserver out of the testing process, but I'd like to keep the function, and constructs in the code for future reference. According to the doc, the client should verify, that whatever traits object it receives conform to a schema in the schema_id and schema_url parameters.

Reply:
I would think that the schema would be served from Kratos